lib/kubeclient/config.rb

```diff
@@ -0,0 +1,113 @@
+require 'yaml'
+require 'base64'
+
+module Kubeclient
+  # Kubernetes client configuration class
+  class Config
+    # Kubernetes client configuration context class
+    class Context
+      attr_reader :api_endpoint, :api_version, :ssl_options
+
+      def initialize(api_endpoint, api_version, ssl_options)
+        @api_endpoint = api_endpoint
+        @api_version = api_version
+        @ssl_options = ssl_options
+      end
+    end
+
+    def initialize(kcfg, kcfg_path)
+      @kcfg = kcfg
+      @kcfg_path = kcfg_path
```

Comment:
@abonas because the configuration file and where the external files (e.g. CA) are stored, could be different paths.